src/Stratis.Bitcoin/Base/ChainState.cs

```diff
@@ -14,6 +14,9 @@ public interface IChainState
         /// <summary>The highest stored block in the repository or <c>null</c> if block store feature is not enabled.</summary>
         ChainedHeader BlockStoreTip { get; set; }
 
+        /// <summary>The highest stored block in the block store cache or <c>null</c> if block store feature is not enabled.</summary>
+        ChainedHeader BlockStoreCacheTip { get; set; }
```

Comment:
I don't think it should go to chain state since we will get rid of chainstate.

Also since this tip is used only in BSBehavior which is in the same namespace we can already add this to BSQ itself instead of adding it to chainstate.

Reply:
>Are we positive that chains state will go away? ConcurrentChain will go for sure yes but why chain state?

Because it's a global state that breaks the architecture.
Every component can easily change any value inside on chainState even though it's not supposed to. 

It was added to as workaround for the problem that some projects can't reference each other but still need to share a state. This problem will soon be solved. After it's solved I see no justification for chain state. 

>This was the smallest change possible in terms of number of lines to change, if I inject store I'd need to change many tests and still need to add the caching property to store.

Yes, implementing it in a way that improves the architecture instead of breaking it will take more time and lines of code, but it's worth it in a long run.